plugins/speculation-rules/hooks.php

```diff
@@ -58,3 +58,75 @@ function plsr_render_generator_meta_tag(): void {
 	echo '<meta name="generator" content="speculation-rules ' . esc_attr( SPECULATION_RULES_VERSION ) . '">' . "\n";
 }
 add_action( 'wp_head', 'plsr_render_generator_meta_tag' );
+
+/**
+ * Filters the HTML output of the search form to inject speculative loading interactivity.
+ *
+ * @since n.e.x.t
+ *
+ * @param string|mixed $form The search form HTML output.
+ * @return string Filtered HTML.
+ */
+function plsr_filter_searchform( $form ): string {
+	if ( ! is_string( $form ) ) {
+		return '';
+	}
+
+	$namespace        = 'speculationRules';
+	$directive_prefix = 'data-wp-on'; // TODO: Use data-wp-on-async when available.
+
+	$p = new WP_HTML_Tag_Processor( $form );
+	while ( $p->next_tag() ) {
+		if ( 'FORM' === $p->get_tag() ) {
+			if ( ! $p->get_attribute( 'data-wp-interactive' ) ) {
+				$p->set_attribute( 'data-wp-interactive', $namespace );
+			}
+			// Create context if not already present.
+			// TODO: Should there be namespaced context?
+			if ( ! $p->get_attribute( 'data-wp-context' ) ) {
+				$p->set_attribute( 'data-wp-context', '{}' );
+			}
+			// TODO: Shouldn't we just watch changes to the one context key?
```

Comment:
I believe this is solved, isn't it?

https://wordpress.slack.com/archives/C071CRKGKUP/p1718106180146629

Reply:
Yes, thanks for following up.

```suggestion
			// Note: This directive only subscribes to the properties that are accessed during its execution.
```